include/wx/hwinfo.h

```diff
@@ -0,0 +1,40 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/hwinfo.h
+// Purpose:     declaration of the wxSystemHardwareInfo class
+// Author:      Blake Madden
+// Created:     2024-07-09
+// Copyright:   (c) 2024 Blake Madden
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_SYSTEMHWINFO_H_
+#define _WX_SYSTEMHWINFO_H_
+
+#include "wx/string.h"
+#include "wx/utils.h"
+#include "wx/power.h"
+
+// ----------------------------------------------------------------------------
+// wxSystemHardwareInfo
+// ----------------------------------------------------------------------------
+
+// Information about the system's hardware
+class WXDLLIMPEXP_BASE wxSystemHardwareInfo
+{
+public:
+
+    static int GetCPUCount();
+
+    static wxString GetCPUArchitectureName();
+    static wxString GetNativeCPUArchitectureName();
+    static bool IsRunningNatively();
+
+    static wxMemorySize GetPhysicalMemory();
+    static wxMemorySize GetFreePhysicalMemory();
+
+    static wxBatteryState GetBatteryState();
+    static int GetRemainingBatteryLifePercent();
+    static wxPowerType GetPowerType();
```

Comment:
I'm not sure we really want to include these functions here, they don't really seem to fit ("power type" is not quite a hardware characteristic) and I think we might provide better API in some `wxBattery` class in `wx/power.h` instead.

Let's keep things minimal for now (it's always simpler to add than to remove them later) and omit this.